app/controllers/admin/source_files_controller.rb

```diff
@@ -45,14 +60,26 @@ def query_template
         .push("LOWER(standards_imports.organization) LIKE ?")
         .push("LOWER(users.name) LIKE ?")
         .push("source_files.public_document = ?")
-        .push("status = ?")
         .join(" OR ")
+        .then { add_status_query(_1) }
```

Comment:
This isn't a totally necessary refactor but I think addresses one bit of potential confusion with the query in combination with our "default scope" above. Without this, what ends up in the query if you're not searching for anything (or searching for something other than a status) is `OR status = NULL`. That's actually fine since there's a null constraint on `status` so it's not really possible for it to be null. But still it felt a little weird to say `WHERE status != 'archived' OR status = NULL`.

To deal with that I just conditionally add the `OR status = ?` part based on whether the user passed in a valid status search. That also required conditionally adding the value below, since otherwise the number of values passed in wouldn't match the number of `?`'s present in the query. What do y'all think?

Reply:
@jutonz I'm excited about this but haven't used `.then` before and am having trouble searching for it. Do you have a reference link?

Reply:
Oh nevermind, this is on the Array object! I was thinking it was on Active Record. Found the info.